packages/api/src/external/carequality/document/process-outbound-document-retrieval-resps.ts

```diff
@@ -231,6 +250,10 @@ async function handleDocReferences(
       };
       transactionBundle.entry?.push(transactionEntry);
 
+      if (!mergedFHIRDocRef.contained || mergedFHIRDocRef.contained.length === 0) {
+        mergedFHIRDocRef.contained = [generateOrganization(cqOrganization.name)];
+      }
+
```

Comment:
this sets the organization to the entry from `cq_directory_entry` in case `authorInstitution` was absent / errored to map

Reply:
This doesn't look very robust: if we have a contained that's not of type `Organization`, we won't add it.

Wouldn't it be better to pass `cqOrganization` as a parameter to `cqToFHIR` and have the logic to add it to the resulting FHIR doc ref there?

Reply:
💯 good point